titiler/xarray/factory.py

```diff
@@ -309,13 +311,42 @@ def tilejson_endpoint(  # type: ignore
                     [-180, -90, 180, 90], list(src_dst.geographic_bounds)
                 )
                 bounds = [max(minx), max(miny), min(maxx), min(maxy)]
+
                 return {
                     "bounds": bounds,
                     "minzoom": minzoom if minzoom is not None else src_dst.minzoom,
                     "maxzoom": maxzoom if maxzoom is not None else src_dst.maxzoom,
                     "tiles": [tiles_url],
                 }
 
+        @self.router.get(
+            "/histogram",
+            response_class=JSONResponse,
+            responses={200: {"description": "Return histogram for this data variable"}},
+            response_model_exclude_none=True,
+        )
+        def histogram(
+            url: str = Query(..., description="Dataset URL"),
+            variable: str = Query(..., description="Variable"),
+            reference: bool = Query(
+                False,
+                title="reference",
+                description="Whether the src_path is a kerchunk reference",
+            ),
+        ):
+            with self.reader(url, variable=variable, reference=reference) as src_dst:
+                boolean_mask = ~np.isnan(src_dst.input)
+                data_values = src_dst.input.values[boolean_mask]
+                counts, values = np.histogram(data_values, bins=10)
```

Comment:
here we may want to update this in the future (when we update titiler version) to 

```python

data = ImageData(src_dst.input.values, mask=boolean_mask)
return data.statistics(
    categorical=True,
    hist_options={"bins": 10},
)
```

Reply:
created https://github.com/developmentseed/titiler-xarray/issues/27